api_tests/harness.js

```diff
@@ -115,49 +106,63 @@ process.once("SIGINT", function(code) {
 // Start services helpers //
 // ********************** //
 
-async function startDockerContainers(names) {
-    await execSync("docker-compose up -d " + names, docker_compose_options);
+async function startDockerContainers(ledgers, ledgers_config) {
+    if (ledgers.length === 0) {
+        throw new Error("No ledgers to start");
+    }
 
-    const subprocess = await spawn(
-        "docker-compose",
-        ["logs", "--tail=all", "-f"],
-        {
-            cwd: docker_cwd,
-            encoding: "utf-8",
-            stdio: [
-                "ignore",
-                fs.openSync(log_dir + "/docker-compose.log", "w"),
-                fs.openSync(log_dir + "/docker-compose-err.log", "w"),
-            ],
-        }
+    let names = [];
+    ledgers.forEach(function(ledger) {
+        names.push(ledgers_config[ledger].docker);
+    });
+    await execSync(
+        "docker-compose up -d " + names.join(" "),
+        docker_compose_options
     );
-    subprocesses.push(subprocess);
+
+    return await spawn("docker-compose", ["logs", "--tail=all", "-f"], {
+        cwd: docker_cwd,
+        encoding: "utf-8",
+        stdio: [
+            "ignore",
+            fs.openSync(log_dir + "/docker-compose.log", "w"),
+            fs.openSync(log_dir + "/docker-compose-err.log", "w"),
+        ],
+    });
+}
+
+async function generateBlock(ledgers, ledgers_config) {
+    if (ledgers) {
+        ledgers.forEach(function(ledger) {
+            if (ledger === "bitcoin") {
+                const block_time_ms = ledgers_config[ledger].block_time_ms;
```

Comment:
❓  What's the point of `block_time_ms`. This looks like a dead variable.

Reply:
yes indeed, fixed.